app/routes/companyInterest/components/CompanyInterestList.tsx

```diff
@@ -84,11 +97,54 @@ class CompanyInterestList extends Component<Props, State> {
         },
       })
       .then(() => {
-        this.props.push(`/companyInterest?semesters=${clickedOption.id}`);
+        this.props.push(
+          `/companyInterest?semesters=${clickedOption.id}&event=${this.props.selectedEventOption.value}`
+        );
       });
   };
+  handleEventChange = (clickedOption: EventOptionType): void => {
+    this.props.push(
+      `/companyInterest?semesters=${this.props.selectedSemesterOption.id}&event=${clickedOption.value}`
+    );
+  };
+
+  componentDidUpdate(prevProps) {
+    if (
+      this.props.selectedSemesterOption !== prevProps.selectedSemesterOption
+    ) {
+      this.setState({
+        generatedCSV: undefined,
+      });
+    }
+  }
+
+  async exportInterestList(event?: string) {
+    const blob = await fetch(
+      getCsvUrl(
+        this.props.selectedSemesterOption.year,
+        this.props.selectedSemesterOption.semester,
+        event
+      ),
+      {
+        headers: {
+          Authorization: `Bearer ${this.props.authToken}`,
+        },
+      }
+    ).then((response) => response.blob());
+    return {
+      url: URL.createObjectURL(blob),
+      filename: `company-interests-${this.props.selectedSemesterOption.year}-${
+        this.props.selectedSemesterOption.semester
+      }${
+        this.props.selectedEventOption.value
+          ? `-${this.props.selectedEventOption.value}`
+          : ''
+      }.csv`,
```

Comment:
Don't you already create the filename in the backend? Why not use that instead here?
Not a huge deal but just a bit unnecessary

Reply:
Truthfully, I didn't really understand the point of having it in the back-end. U have to have this here in the front-end for the purpose of changing url blobs.